### PR TITLE
[Spring] Add new params to support auto binding

### PR DESCRIPTION
**Pull request by moarychan**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

* Add arguments `--bind-service-registry` in `spring app create`.
* Add arguments `--bind-application-configuration-service` in `spring app create`.

Instead of PR https://github.com/Azure/azure-cli-extensions/pull/7083.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

File: src/spring/azext_spring/_app_factory.py [hunk 1]
```diff
@@ -175,6 +176,17 @@ def _load_secrets_config(self, secrets=None, **_):
...
         return secret_var_def
 
+    def _load_addon_configs(self, bind_service_registry=None, bind_application_configuration_service=None, **_):
+        if not bind_service_registry and not bind_application_configuration_service:
+            return None
+
+        addon_configs = {}
+
+        if bind_service_registry:
+             addon_configs['serviceRegistry'] = {'resourceId': bind_service_registry}
```
Comment (yuwzho) [inline src/spring/azext_spring/_app_factory.py:186]:
```suggestion
            addon_configs['serviceRegistry'] = {'resourceId': bind_service_registry}
```

File: src/spring/azext_spring/_app_factory.py [hunk 2]
```diff
@@ -175,6 +176,17 @@ def _load_secrets_config(self, secrets=None, **_):
...
+        if bind_application_configuration_service:
+             addon_configs['applicationConfigurationService'] = {'resourceId': bind_application_configuration_service}
```
Comment (yuwzho) [inline src/spring/azext_spring/_app_factory.py:188]:
```suggestion
            addon_configs['applicationConfigurationService'] = {'resourceId': bind_application_configuration_service}
```

File: src/spring/azext_spring/_app_factory.py [hunk 3]
```diff
@@ -175,6 +176,17 @@ def _load_secrets_config(self, secrets=None, **_):
...
+        if not bind_service_registry and not bind_application_configuration_service:
+            return None
+
+        addon_configs = {}
+
+        if bind_service_registry:
+            addon_configs['serviceRegistry'] = {'resourceId': bind_service_registry}
+        if bind_application_configuration_service:
+            addon_configs['applicationConfigurationService'] = {'resourceId': bind_application_configuration_service}
+        return addon_configs
```
Comment (yuwzho) [inline src/spring/azext_spring/_app_factory.py:189]:
```suggestion
        return addon_configs

```

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by moarychan** [comment]
@yuwzho  @zhoxing-ms PR is ready for review, the check style failures can be ignored as they can be ignored and will not prevent the PR from being merged. Many thanks!

**Review by yuwzho — APPROVED** [review]
LGTM. The failed CI is warning about the original code. Let us keep this PR clean for the new feature and we commit the incremental code passed all CI check.

**Comment by moarychan** [comment]
Hi @zhoxing-ms @yonzhan, please help review, Many thanks!

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**